packages/fxa-shared/lib/sentry.ts

```diff
@@ -0,0 +1,175 @@
+import Logger from './logger';
+
+const ALLOWED_QUERY_PARAMETERS = [
+  'automatedBrowser',
+  'client_id',
+  'context',
+  'entrypoint',
+  'keys',
+  'migration',
+  'redirect_uri',
+  'scope',
+  'service',
+  'setting',
+  'style',
+];
+
+/**
+ * function that gets called before data gets sent to error metrics
+ *
+ * @param {Object} data
+ *  Error object data
+ * @returns {Object} data
+ *  Modified error object data
+ * @private
+ */
+function beforeSend(data: Sentry.Event): Sentry.Event {
+  if (data.request) {
+    if (data.request.url) {
+      data.request.url = cleanUpQueryParam(data.request.url);
+    }
+
+    if (data.tags) {
+      // if this is a known errno, then use grouping with fingerprints
+      // Docs: https://docs.sentry.io/hosted/learn/rollups/#fallback-grouping
+      if (data.tags.errno) {
+        data.fingerprint = ['errno' + data.tags.errno];
+        // if it is a known error change the error level to info.
+        data.level = Sentry.Severity.Info;
+      }
+    }
+
+    if (data.exception?.values) {
+      data.exception.values.forEach(value => {
+        if (value.stacktrace && value.stacktrace.frames) {
+          value.stacktrace.frames.forEach((frame: { abs_path?: string }) => {
+            if (frame.abs_path) {
+              frame.abs_path = cleanUpQueryParam(frame.abs_path); // eslint-disable-line camelcase
+            }
+          });
+        }
+      });
+    }
+
+    if (data.request?.headers?.Referer) {
+      data.request.headers.Referer = cleanUpQueryParam(
+        data.request.headers.Referer
+      );
+    }
+  }
+
+  return data;
+}
+
+/**
+ * Overwrites sensitive query parameters with a dummy value.
+ *
+ * @param {String} url
+ * @returns {String} url
+ * @private
+ */
+function cleanUpQueryParam(url = '') {
+  const urlObj = new URL(url);
+
+  if (!urlObj.search.length) {
+    return url;
+  }
+
+  // Iterate the search parameters.
+  urlObj.searchParams.forEach((_, key) => {
+    if (!ALLOWED_QUERY_PARAMETERS.includes(key)) {
+      // if the param is a PII (not allowed) then reset the value.
+      urlObj.searchParams.set(key, 'VALUE');
+    }
+  });
+
+  return urlObj.href;
+}
+
+/**
+ * Exception fields that are imported as tags
+ */
+const exceptionTags = ['code', 'context', 'errno', 'namespace', 'status'];
+
+interface SentryMetrics {
+  _logger: Logger;
+  configure: (arg0: string, arg1?: string) => void;
+  captureException: (arg0: Error) => void;
+  __beforeSend: (arg0: Sentry.Event) => Sentry.Event;
+  __cleanUpQueryParam: (arg0: string) => string;
+}
+
+/**
+ * Creates a SentryMetrics singleton object that starts up Sentry/browser.
+ *
+ * This must be configured with the `configure` method before use.
+ *
+ * Read more at https://docs.sentry.io/platforms/javascript
+ *
+ * @constructor
+ */
+const SentryMetrics = (function(this: SentryMetrics) {
+  this._logger = new Logger();
+  // tslint:disable-next-line: callable-types
+} as any) as { new (): SentryMetrics };
```

Comment:
Could update to `new () => SentryMetrics` for syntax consistency with the other function constructor above.